packages/database/test/helpers/syncPointSpec.json

```diff
@@ -4179,6 +4179,249 @@
       }
     ]
   },
+  {
+    "name": "Queries with startAfter and endBefore work",
+    "steps":
+    [
+      {
+        "type": "listen",
+        "path": "",
+        "params": {
+          "tag": 1,
+          "orderBy": "index",
+          "startAfter": { "index": 1 },
+          "endBefore": { "index": 10 }
+        },
+        "events": []
+      },
+      {
+        ".comment": "update from server sends all data",
+        "type": "serverUpdate",
+        "path": "",
+        "data": {
+          "a": { "index": 0, "value": "a" },
+          "b": { "index": 2, "value": "b" },
+          "c": { "index": 9, "value": "c" },
+          "d": { "index": 11, "value": "d" }
+        },
+        "events": [
+          {
+            "path": "",
+            "type": "child_added",
+            "name": "b",
+            "prevName": null,
+            "data": { "index": 2, "value": "b" }
+          },
+          {
+            "path": "",
+            "type": "child_added",
+            "name": "c",
+            "prevName": null,
```

Comment:
This should be `b`. Your test should be failing locally. Can you re-merge my branch into yours?

Reply:
The test is actually passing locally on https://github.com/firebase/firebase-js-sdk/pull/6706/commits/331085549fd42a8b679f149c16c65b83be25e4df. It, and the next one, starts failing once I pull https://github.com/firebase/firebase-js-sdk/pull/6786/commits/888a98df4f043f957bf61b0057f89b6233985c4e. I think the previous spec tests are somehow missing events.

Reply:
Perhaps we were accidentally only validating the last one: https://github.com/firebase/firebase-js-sdk/pull/6786/files#diff-872d46e57e4152b3a7bd18b86102a8428117be4d55ee65e287439de27b7ee0efL319.

Reply:
Sorry, I mean when you pull in my branch, it should fail

Reply:
Oop, yeah, it just started failing. Will fix.